torchprime/torch_xla_models/experimental/custom_kernel.py

```diff
@@ -0,0 +1,471 @@
+import dataclasses
+import functools
+import json
+import os
+from collections.abc import Callable
+from contextlib import contextmanager
+from dataclasses import asdict
+from typing import Any
+
+import torch
+import torch_xla
+import torch_xla.debug.profiler as xp
+from torch.library import custom_op
+from torch.utils._pytree import tree_flatten
+from torch_xla.core.xla_builder import jax_func_to_xla_computation
+from torch_xla.distributed.spmd import Mesh
+
+# use the global var to cache XlaComputation
+func_arg_to_XlaComputation: dict[tuple[Any], str] = {}
+
+
+@contextmanager
+def _jax_env_context():
```

Comment:
Could we import this function from torch_xla until everything is eventually upstreamed? I think Bhavya did some fixes to it and it would be nice to keep things in sync.